feat: add GHOSTTY_BIN_DIR to path via shell integration

matthew-hre opened the pull request:
Closes #8956

Elvish written by Copilot, the rest was written by me with AI documentation.

matthew-hre committed:
feat: add GHOSTTY_BIN_DIR to path via shell integration

jparise reviewed (commented):
Nice work! I spotted a few possible improvements while reading through the changes, and I'll come back soon and actually test everything before approving.

matthew-hre committed:
fix: cleaned up elvish and fish integrations for bin_dir

mitchellh reviewed (commented):
Thank you very much for doing this! This will be a great addition to 1.2.1, pending @jparise's approval.

jparise committed:
shell-integration: remove redundant comments

I think the conditions are sufficiently self-descriptive.

jparise committed:
shell-integration: append $GHOSTTY_BIN_DIR to $PATH

For consistency with the termio/Exec.zig implementation, we always
append to the PATH (lowest priority).

jparise reviewed (approved):
Good stuff! I pushed one functional change: we always want to _append_ to `PATH` (lowest priority).
